### ISSUE TITLE
ClassNotFoundException on player join

### ISSUE THREAD

**Issue by SlimeDog** [Zrips/Residence#346]
Spigot 1.13.2 final
or
Spigot 1.14.4 (Jenkins 2435) git-Spigot-e5b1b5d-e5e8eec (MC: 1.14.4) (Implementing API version 1.14.4-R0.1-SNAPSHOT)

Residence 4.8.7.0

On player connect:
```
[15:05:07] [Server thread/ERROR]: Could not pass event PlayerJoinEvent to Residence v4.8.7.0
org.bukkit.event.EventException: null
        at org.bukkit.plugin.java.JavaPluginLoader$1.execute(JavaPluginLoader.java:320) ~[spigot-1.14.4.jar-2019-08-03-1335:git-Spigot-e5b1b5d-e5e8eec]
        at org.bukkit.plugin.RegisteredListener.callEvent(RegisteredListener.java:70) ~[spigot-1.14.4.jar-2019-08-03-1335:git-Spigot-e5b1b5d-e5e8eec]
        at org.bukkit.plugin.SimplePluginManager.fireEvent(SimplePluginManager.java:529) [spigot-1.14.4.jar-2019-08-03-1335:git-Spigot-e5b1b5d-e5e8eec]
        at org.bukkit.plugin.SimplePluginManager.callEvent(SimplePluginManager.java:514) [spigot-1.14.4.jar-2019-08-03-1335:git-Spigot-e5b1b5d-e5e8eec]
        at net.minecraft.server.v1_14_R1.PlayerList.a(PlayerList.java:173) [spigot-1.14.4.jar-2019-08-03-1335:git-Spigot-e5b1b5d-e5e8eec]
        at net.minecraft.server.v1_14_R1.LoginListener.c(LoginListener.java:154) [spigot-1.14.4.jar-2019-08-03-1335:git-Spigot-e5b1b5d-e5e8eec]
        at net.minecraft.server.v1_14_R1.LoginListener.tick(LoginListener.java:53) [spigot-1.14.4.jar-2019-08-03-1335:git-Spigot-e5b1b5d-e5e8eec]
        at net.minecraft.server.v1_14_R1.NetworkManager.a(NetworkManager.java:220) [spigot-1.14.4.jar-2019-08-03-1335:git-Spigot-e5b1b5d-e5e8eec]
        at net.minecraft.server.v1_14_R1.ServerConnection.c(ServerConnection.java:129) [spigot-1.14.4.jar-2019-08-03-1335:git-Spigot-e5b1b5d-e5e8eec]
        at net.minecraft.server.v1_14_R1.MinecraftServer.b(MinecraftServer.java:1091) [spigot-1.14.4.jar-2019-08-03-1335:git-Spigot-e5b1b5d-e5e8eec]
        at net.minecraft.server.v1_14_R1.DedicatedServer.b(DedicatedServer.java:393) [spigot-1.14.4.jar-2019-08-03-1335:git-Spigot-e5b1b5d-e5e8eec]
        at net.minecraft.server.v1_14_R1.MinecraftServer.a(MinecraftServer.java:970) [spigot-1.14.4.jar-2019-08-03-1335:git-Spigot-e5b1b5d-e5e8eec]
        at net.minecraft.server.v1_14_R1.MinecraftServer.run(MinecraftServer.java:815) [spigot-1.14.4.jar-2019-08-03-1335:git-Spigot-e5b1b5d-e5e8eec]
        at java.lang.Thread.run(Thread.java:834) [?:?]
Caused by: java.lang.NoClassDefFoundError: com/Zrips/CMI/Modules/Permissions/PermissionInfo
        at com.bekvon.bukkit.residence.permissions.PermissionManager.addToCache(PermissionManager.java:512) ~[?:?]
        at com.bekvon.bukkit.residence.permissions.PermissionManager$ResPerm.hasPermission(PermissionManager.java:386) ~[?:?]
        at com.bekvon.bukkit.residence.permissions.PermissionManager$ResPerm.hasPermission(PermissionManager.java:361) ~[?:?]
        at com.bekvon.bukkit.residence.permissions.PermissionManager$ResPerm.hasPermission(PermissionManager.java:357) ~[?:?]
        at com.bekvon.bukkit.residence.permissions.PermissionManager$ResPerm.hasPermission(PermissionManager.java:325) ~[?:?]
        at com.bekvon.bukkit.residence.permissions.PermissionManager.isResidenceAdmin(PermissionManager.java:103) ~[?:?]
        at com.bekvon.bukkit.residence.listeners.ResidencePlayerListener.onPlayerJoin(ResidencePlayerListener.java:915) ~[?:?]
        at jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method) ~[?:?]
        at jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62) ~[?:?]
        at jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43) ~[?:?]
        at java.lang.reflect.Method.invoke(Method.java:566) ~[?:?]
        at org.bukkit.plugin.java.JavaPluginLoader$1.execute(JavaPluginLoader.java:316) ~[spigot-1.14.4.jar-2019-08-03-1335:git-Spigot-e5b1b5d-e5e8eec]
        ... 13 more
Caused by: java.lang.ClassNotFoundException: com.Zrips.CMI.Modules.Permissions.PermissionInfo
        at java.net.URLClassLoader.findClass(URLClassLoader.java:471) ~[?:?]
        at org.bukkit.plugin.java.PluginClassLoader.findClass(PluginClassLoader.java:135) ~[spigot-1.14.4.jar-2019-08-03-1335:git-Spigot-e5b1b5d-e5e8eec]
        at org.bukkit.plugin.java.PluginClassLoader.findClass(PluginClassLoader.java:81) ~[spigot-1.14.4.jar-2019-08-03-1335:git-Spigot-e5b1b5d-e5e8eec]
        at java.lang.ClassLoader.loadClass(ClassLoader.java:588) ~[?:?]
        at java.lang.ClassLoader.loadClass(ClassLoader.java:521) ~[?:?]
        at com.bekvon.bukkit.residence.permissions.PermissionManager.addToCache(PermissionManager.java:512) ~[?:?]
        at com.bekvon.bukkit.residence.permissions.PermissionManager$ResPerm.hasPermission(PermissionManager.java:386) ~[?:?]
        at com.bekvon.bukkit.residence.permissions.PermissionManager$ResPerm.hasPermission(PermissionManager.java:361) ~[?:?]
        at com.bekvon.bukkit.residence.permissions.PermissionManager$ResPerm.hasPermission(PermissionManager.java:357) ~[?:?]
        at com.bekvon.bukkit.residence.permissions.PermissionManager$ResPerm.hasPermission(PermissionManager.java:325) ~[?:?]
        at com.bekvon.bukkit.residence.permissions.PermissionManager.isResidenceAdmin(PermissionManager.java:103) ~[?:?]
        at com.bekvon.bukkit.residence.listeners.ResidencePlayerListener.onPlayerJoin(ResidencePlayerListener.java:915) ~[?:?]
        at jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method) ~[?:?]
        at jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62) ~[?:?]
        at jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43) ~[?:?]
        at java.lang.reflect.Method.invoke(Method.java:566) ~[?:?]
        at org.bukkit.plugin.java.JavaPluginLoader$1.execute(JavaPluginLoader.java:316) ~[spigot-1.14.4.jar-2019-08-03-1335:git-Spigot-e5b1b5d-e5e8eec]
        ... 13 more
```

**Comment by SlimeDog** [comment]
Is there supposed to be CMI code in Residence?

**Comment by SlimeDog** [comment]
This issue still exists in Residence 4.8.7.1.

**Comment by SlimeDog** [comment]
Confirmed fixed in Residence 4.8.7.2 with Spigot 1.14.4
No longer testing Spigot 1.13.2, now that Spigot 1.14.4 is official release
